recycleIT-frontend/src/Components/Athorization/Authorization.tsx

```diff
@@ -30,48 +32,87 @@ interface IFormProps {
 
 // minimum eight characters, at least one letter and one number and no spaces
 const passwordRegexpr = /^(?=.*[A-Za-z])(?=.*\d)(?=^\S*$)[A-Za-z\d]{8,}$/
-
 const emailRegexpr = /^([\w.%+-]+)@([\w-]+\.)+([\w]{2,})$/i
 
 const Authorization = () => {
+    const user = useContext(UserContext);
     const [values, setValues] = useState<IFormState>({
         name: '',
         email: '',
         password: '',
         repeatPassword: '',
         showPassword: false,
-        isFormValid: false
     });
     const [validationError, setValidationError] = useState({
         name: false,
         email: false,
         password: false,
         passwordsMatch: false,
+        isFormInvalid: false
     });
 
-    const [isLoginMode, setIsLoginMode] = useState(false);
+    const [isLoginMode, setIsLoginMode] = useState(true);
+    const [error, setError] = useState(false);
+    const [loading, setLoading] = useState(false);
+    const [errorMessage, setErrorMessage] = useState(user?.errorMessage || '');
+    const navigate = useNavigate();
+
+    if (user?.isLoggedIn) {
+        navigate(-1);
+    }
+
+    useEffect(() => {
+        console.log(user);
+        if (user?.loading) {
+            setLoading(true)
+            setError(false)
+        } else { 
+            setLoading(false)
+        }
+        if (user?.isRegistered) {
+            setIsLoginMode(true)
+        } 
+        if (user?.error) {
+            setError(true);
+            setErrorMessage(user.errorMessage);
+        }
+    }, [user])
+
+    useEffect(() => {
+        setValues({
+            name: '', 
+            email: '', 
+            password: '', 
+            repeatPassword: '',
+            showPassword: false
+        })
+    }, [isLoginMode])
 
     const validateForm = (name: string, email: string, password: string, repeatPassword: string) => {
```

Comment:
When seeing a function validateForm you would expect it to return true if form is valid and false if form is invalid, this would also make submitForm function more readable